modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheMapEntry.java

```diff
@@ -5088,14 +5088,7 @@ private void unlockListenerReadLock() {
 
     /** {@inheritDoc} */
     @Override public String toString() {
-        lockEntry();
-
-        try {
-            return S.toString(GridCacheMapEntry.class, this);
-        }
-        finally {
-            unlockEntry();
-        }
+        return S.toString(GridCacheMapEntry.class, this);
```

Comment:
Are you sure we haven't introduced a risk of ConcurrentModificationException?

GridCacheMapEntry:
    /** Extras */
    @GridToStringInclude
    private GridCacheEntryExtras extras;

GridCacheMvcc:
    /** Local queue. */
    @GridToStringInclude
    private LinkedList<GridCacheMvccCandidate> locs;

Access to candidates list becomes unsynchronized.